### PR TITLE
chore(IDX): Add hotfix and release logic

### PR DESCRIPTION
**Pull request by cgundy**
The main CI jobs need to also run on hotfix and rc branches.

Additionally, we need to run a separate CI job that runs different tests for hotfixes.

### CODE REVIEW

File: .github/workflows-source/ci-main.yml [hunk 1]
```diff
@@ -64,11 +68,18 @@ jobs:
   bazel-test-all:
     name: Bazel Test All
     <<: *dind-large-setup
-    env:
-      TITLE: ${{ github.event.pull_request.title }}
     steps:
       - <<: *checkout
       - <<: *before-script
+      - name: check Trigger type
```
Comment (marko-k0) [inline .github/workflows-source/ci-main.yml]:
```suggestion
      - name: Check trigger type
```

File: .github/workflows-source/ci-main.yml [hunk 2]
```diff
@@ -64,11 +68,18 @@ jobs:
...
+        run: |
+          if [[ "${{ github.event.pull_request.labels.*.name }}" == *"merge-train"* ]]; then
```
Comment (marko-k0) [inline .github/workflows-source/ci-main.yml:77]: There are "merge-train" labels on GitHub?

Comment (cgundy) [reply .github/workflows-source/ci-main.yml:78]: Let me see if there's a better option...

File: .github/workflows-source/ci-main.yml [hunk 3]
```diff
@@ -64,11 +68,21 @@ jobs:
   bazel-test-all:
     name: Bazel Test All
     <<: *dind-large-setup
-    env:
-      TITLE: ${{ github.event.pull_request.title }}
     steps:
       - <<: *checkout
       - <<: *before-script
+      - name: check trigger type
```
Comment (marko-k0) [inline .github/workflows-source/ci-main.yml:74]: It's good that we're consistent. If all other names start with capita, it makes sense to have "Check trigger type" here.

Comment (marko-k0) [reply .github/workflows-source/ci-main.yml]: And maybe better to name it "Setting BAZEL_EXTRA_ARGS_RULES".

File: .github/workflows-source/ci-main.yml [hunk 4]
```diff
@@ -7,6 +7,10 @@ on:
   schedule:
     - cron: "0 * * * *"
   pull_request:
+  push:
+      branches:
```
Comment (marko-k0) [inline .github/workflows-source/ci-main.yml:11]: You should fix indention for the sake of consistency.

**Review by marko-k0 — APPROVED** [review]

**Pull request merged**